src/main/java/org/broadinstitute/hellbender/tools/copynumber/PostprocessGermlineCNVCalls.java

```diff
@@ -325,6 +347,48 @@ private void generateIntervalsVCFFileFromAllShards() {
+                final File shardRootDirectory = sortedCallsShardPaths.get(shardIndex);
+                final File denoisedCopyRatioFile = getSampleDenoisedCopyRatioFile(shardRootDirectory, sampleIndex);
+                final DenoisedCopyRatioCollection shardDenoisedCopyRatioCollectionForShard = new DenoisedCopyRatioCollection(denoisedCopyRatioFile);
+                final List<SimpleInterval> shardIntervals = sortedIntervalCollections.get(shardIndex).getIntervals();
+                final String sampleNameFromDenoisedCopyRatioFile = shardDenoisedCopyRatioCollectionForShard
+                        .getMetadata().getSampleName();
+                Utils.validate(sampleNameFromDenoisedCopyRatioFile.equals(sampleName),
+                        String.format("Sample name found in the header of denoised copy ratio file for shard %d " +
+                                        "different from the expected sample name (found: %s, expected: %s).",
+                                shardIndex, sampleNameFromDenoisedCopyRatioFile, sampleName));
+                final List<LinearNonLocatableCopyRatio> shardDenoisedCopyRatioRecords = shardDenoisedCopyRatioCollectionForShard.getRecords();
+                Utils.validate(shardIntervals.size() == shardDenoisedCopyRatioRecords.size(),
+                        String.format("The number of entries in denoised copy ratio file for shard %d does " +
+                                        "not match the number of entries in the shard interval list (copy ratio list size: %d, " +
+                                        "interval list size: %d)", shardIndex, shardDenoisedCopyRatioRecords.size(),
+                                shardIntervals.size()));
+                concatenatedIntervalList.addAll(shardIntervals);
+                concatenatedDenoisedCopyRatioRecordsList.addAll(shardDenoisedCopyRatioRecords);
+            }
+            /* Attach the corresponding intervals */
+            final List<DenoisedLocatableCopyRatio> denoisedLocatableCopyRatioList =
+                    IntStream.range(0, concatenatedIntervalList.size()).mapToObj(intervalIndex ->
+                            new DenoisedLocatableCopyRatio(concatenatedIntervalList.get(intervalIndex),
+                                    concatenatedDenoisedCopyRatioRecordsList.get(intervalIndex))).collect(Collectors.toList());
+            final SimpleSampleLocatableMetadata metadata = new SimpleSampleLocatableMetadata(sampleName, sequenceDictionary);
+            /* Make a locatable collection of denoised copy ratios and write it to file */
+            final DenoisedLocatableCopyRatioCollection denoisedLocatableCopyRatioCollection =
+                    new DenoisedLocatableCopyRatioCollection(metadata, denoisedLocatableCopyRatioList);
+            denoisedLocatableCopyRatioCollection.write(outputDenoisedCopyRatioFile);
+            logger.info("Finished writing the denoised copy ratios file.");
```

Comment:
To be consistent with the log messages for VCF output, should be `logger.info(String.format("Writing denoised copy ratios to %s...", outputDenoisedCopyRatioFile.getAbsolutePath()));` and should precede the actual write statement.

Reply:
Done